src/connector/src/source/common.rs

```diff
@@ -74,3 +76,31 @@ pub(crate) async fn into_chunk_stream(
         yield msg_batch?;
     }
 }
+
+#[derive(Clone, Debug, Deserialize, PartialEq, with_options::WithOptions)]
+pub struct SecretString {
+    inner: redact::Secret<String>,
+}
```

Comment:
What about `#[serde(transparent)]`?

Reply:
Serialize need be implemented for `redact::Secret<String>` manually.

Reply:
`redact::Secret` doesn't implement `Serialize`.

Reply:
Meanwhile `impl Serialize for SecretString` should be chosen according to use case, i.e. whether the original value or the redacted on.

Though currently there is not use case that serializes `SecretString`.